onnx/defs/nn/defs.cc

```diff
@@ -292,6 +292,75 @@ computes the output.)DOC";
         "number of groups input channels and output channels are divided into, default is 1.",
         AttributeProto::INT,
         static_cast<int64_t>(1));
+    schema.TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+        propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+        if (!hasNInputShapes(ctx, 1)) {
+          return;
+        }
+
+        // don't bother with legacy auto_pad for now
+        if (ctx.getAttribute("auto_pad")) {
+          return;
+        }
+
+        std::vector<int64_t> dilations;
+        if (!getRepeatedAttribute(ctx, "dilations", dilations)) {
+          for (int i = 0; i < ctx.getInputType(0)->tensor_type().shape().dim_size(); ++i) {
+            dilations.push_back(1);
+          }
+        }
+
+        std::vector<int64_t> kernel_shape;
+        if (!getRepeatedAttribute(ctx, "kernel_shape", kernel_shape)) {
+          for (int i = 2; i < ctx.getInputType(1)->tensor_type().shape().dim_size(); ++i) {
+            kernel_shape.push_back(ctx.getInputType(1)->tensor_type().shape().dim(i).dim_value());
+          }
+        }
+
+        std::vector<int64_t> pads;
+        if (!getRepeatedAttribute(ctx, "pads", pads)) {
+          for (int i = 0; i < ctx.getInputType(0)->tensor_type().shape().dim_size(); ++i) {
+            pads.push_back(0);
+            pads.push_back(0);
+          }
+        }
+
+        std::vector<int64_t> strides;
+        if (!getRepeatedAttribute(ctx, "strides", strides)) {
+          for (int i = 0; i < ctx.getInputType(0)->tensor_type().shape().dim_size(); ++i) {
+            strides.push_back(1);
+          }
+        }
+
```

Comment:
Could you add the formula here: how do you calculate the output shape...

Reply:
it's in the inline comments below